sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFs.scala

```diff
@@ -191,18 +157,18 @@ private[hive] case class HiveGenericUDF(
 
   override protected def withNewChildrenInternal(newChildren: IndexedSeq[Expression]): Expression =
     copy(children = newChildren)
-  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+
+  protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
     val refTerm = ctx.addReferenceObj("this", this)
```

Comment:
do we still reference the entire `this`? Can we just reference the new evalautor?

Reply:
Very good suggestions!
BTW, when we reference the new evaluator, some properties in `HiveGenericUDFEvaluator` will have to use lazy, because it needs to be recreated on the Executor side.